lib/index.d.ts

```diff
@@ -0,0 +1,161 @@
+import ascend from "./ascend";
+import az from "./az";
+import both from "./both";
+import { call } from "when/node";
+
+declare let nanoutils: Nanoutils.Static
+
+declare namespace Nanoutils {
+  type Predicate1<T1> = (value: T1) => boolean
+
+  // addIndex
+  interface ArrayIterator<T1, T2> {
+    (change: (value: T1) => T2, array: ReadonlyArray<T1>): T2[]
```

Comment:
This restricts a return value as it should always be of a type `T2[]` however, it can be different, i.e.

<img width="888" alt="screen shot 2018-10-08 at 01 04 48" src="https://user-images.githubusercontent.com/2991847/46587513-5da80e00-ca96-11e8-8247-e63268606ddc.png">

So better here use `any[]`